data/kernels/filmic.cl

```diff
@@ -104,3 +113,173 @@ filmic (read_only image2d_t in, write_only image2d_t out, int width, int height,
+  const float4 sat = saturation;
+  return lum + sat * (x - lum);
+}
+
+
+kernel void
+filmicrgb_split (read_only image2d_t in, write_only image2d_t out,
+                 int width, int height,
+                 const float dynamic_range, const float black_exposure, const float grey_value,
+                 global const dt_colorspaces_iccprofile_info_cl_t *profile_info,
+                 read_only image2d_t lut, const int use_work_profile,
+                 const float sigma_toe, const float sigma_shoulder, const float saturation,
+                 const float4 M1, const float4 M2, const float4 M3, const float4 M4, const float4 M5,
+                 const float latitude_min, const float latitude_max, const float output_power)
+{
+  const unsigned int x = get_global_id(0);
+  const unsigned int y = get_global_id(1);
+
+  if(x >= width || y >= height) return;
+
+  const float4 i = read_imagef(in, sampleri, (int2)(x, y));
+  float4 o;
+
+  const float4 noise4 = native_powr(2.0f, -16.0f);
+  const float4 dynamic4 = dynamic_range;
+  const float4 blacks4 = black_exposure;
+  const float4 grey4 = grey_value;
+
+  // Log tonemapping
+  o = (i < noise4) ? noise4 : i;
+  o = (native_log2(o / grey4) - blacks4) / dynamic4;
+  o = clamp(o, noise4, (float4)1.0f);
+
+  // Selective desaturation of extreme luminances
+  const float luminance = (use_work_profile) ? get_rgb_matrix_luminance(o, profile_info, lut) : dt_camera_rgb_luminance(o);
+  const float desaturation = filmic_desaturate(luminance, sigma_toe, sigma_shoulder, saturation);
+  o = linear_saturation(o, luminance, desaturation);
+
+  // Filmic spline
+  o.x = filmic_spline(o.x, M1, M2, M3, M4, M5, latitude_min, latitude_max);
+  o.y = filmic_spline(o.y, M1, M2, M3, M4, M5, latitude_min, latitude_max);
+  o.z = filmic_spline(o.z, M1, M2, M3, M4, M5, latitude_min, latitude_max);
+
+  // Output power
+  o = native_powr(clamp(o, (float4)0.0f, (float4)1.0f), output_power);
+
+  // Copy alpha layer and save
+  o.w = i.w;
+  write_imagef(out, (int2)(x, y), o);
+}
+
+
+inline float pixel_rgb_norm_power(const float4 pixel)
+{
+  // weird norm sort of perceptual. This is black magic really, but it looks good.
+  float4 RGB = fabs(pixel);
+  RGB.w = 0.0f; // ensure we don't count the alpha channel in the norm
+  const float4 RGB_square = RGB * RGB;
+  const float4 RGB_cubic = RGB_square * RGB;
+  const float4 ones = 1.0f;
+  return dot(RGB_cubic, ones)/ dot(RGB_square, ones);
+}
+
+
+inline float get_pixel_norm(const float4 pixel, const dt_iop_filmicrgb_methods_type_t variant,
+                            global const dt_colorspaces_iccprofile_info_cl_t *const work_profile,
+                            read_only image2d_t lut, const int use_work_profile)
+{
+
+  switch(variant)
+  {
+    case(DT_FILMIC_METHOD_MAX_RGB):
+      return fmax(fmax(pixel.x, pixel.y), pixel.z);
+
+    case(DT_FILMIC_METHOD_LUMINANCE):
+      return (use_work_profile) ? get_rgb_matrix_luminance(pixel, work_profile, lut)
```

Comment:
this return could be replaced with:

`return dt_rgb_norm(pixel, DT_RGB_NORM_LUMINANCE, use_work_profile, work_profile, lut);`

To avoid code duplication.

Reply:
Note that the "luminance Y" issue reported is when user_work_profile is 1 and so I suspect something wrong with the initialization of work_profile or lut parameter.